src/components/Web3Provider/index.tsx

```diff
@@ -15,7 +18,37 @@ export default function Web3Provider({ children }: { children: ReactNode }) {
 
   return (
     <Web3ReactProvider connectors={connectors} key={key}>
+      <Tracer />
       {children}
     </Web3ReactProvider>
   )
 }
+
+function Tracer() {
```

Comment:
can't we just make this a hook instead a full component?

Reply:
We can't access the Web3ReactProvider's context without being a nested JSXElement.
Hooks won't have access to `useWeb3React().provider`, which is necessary to get a (potentially third-party) active provider.

Reply:
ah yea you're right